### PR TITLE
adding timeout for get roles with constants

### PR DESCRIPTION
**Pull request by cviecco**

### CODE REVIEW

File: broker/aws/impl.go [hunk 1]
```diff
@@ -283,30 +283,40 @@ func (b *Broker) withProfileAssumeRole(accountName string, profileName string,
...
+				return !lastPage
+			})
+		c <- err
+	}()
+	select {
+	case getRolesErr := <-c:
+		if getRolesErr != nil {
+			return nil, getRolesErr
+		}
+		awsListRolesSuccess.WithLabelValues(accountName).Inc()
```
Comment (cviecco) [inline broker/aws/impl.go:313]: Maybe I should move this out of the select?

File: broker/aws/impl.go [hunk 2]
```diff
@@ -283,30 +283,40 @@ func (b *Broker) withProfileAssumeRole(accountName string, profileName string,
...
-			b.logger.Debugf(2, "listrolesoutput =%v", listRolesOutput)
-			b.logger.Debugf(2, "loop, lastPage=%s", lastPage)
-			for _, role := range listRolesOutput.Roles {
-				roleNames = append(roleNames, *role.RoleName)
-			}
-			return !lastPage
-		})
-	if err != nil {
-		return nil, err
+	c := make(chan error)
```
Comment (rgooch) [inline broker/aws/impl.go:294]: Please make the capacity equal to the number of accounts to iterate over.

File: broker/aws/impl.go (hunk 1 above)
Comment (rgooch) [reply broker/aws/impl.go:313]: That's probably cleaner.

File: broker/aws/impl.go (hunk 2 above)
Comment (rgooch) [reply broker/aws/impl.go:294]: In this case, capacity should be 1.
As a general rule, channels should not be blocking (capacity=0) unless there is a good reason.

Comment (cviecco) [reply broker/aws/impl.go]: thank you fixes

File: broker/aws/impl.go (hunk 1 above)
Comment (cviecco) [reply broker/aws/impl.go]: done

**Pull request merged**